lib/logger.js

```diff
@@ -19,6 +17,9 @@ function Logger () {
 }
 
 Logger.prototype.setLogLevel = function (logLevel) {
```

Comment:
If there is a `setLogLevel` there should be a `getLogLevel`.
Even better create getter, setter for logLevel and save the value in `_logLevel`.
For saving compatibility with older why wrap this function with `deprecated` (https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_util_deprecate_fn_msg_code)

Reply:
When saying `getter` and `setter` you're talking about simply functions that set/return the value of the `_logLevel` field, right? Because in that case, the deprecated property is not `setLogLevel` but rather `logLevel`.
In any case, I'm not sure this is in the scope of thie PR. I mean, if I wanted to completly clean this I would also change this into an ES6 class. I didn't mean to change the code any more then I needed to copmlete the expected behaviour of the existing API.

Reply:
I mean a JS getter/setter : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get.
It might not be in the scope of the PR but we already added a new method `getLogLevel`, so we do add api which we (or a least me) think it's not good. Also, making this an ES6 class also don't take a lot of time so for such small class.

Anyway, I will leave this call for you.